in_situ/swe/swe_decom.py

```diff
@@ -0,0 +1,20 @@
+from pathlib import Path
+from space_packet_parser import xtcedef, parser
+import bitstring
+
+packet_file = Path('science_block_20221116_163611Z_idle.bin')
+with packet_file.open(mode='rb') as fh:
+    binary_data = bitstring.ConstBitStream(fh)
+
+xtce_document = Path('swe_packet_definition.xml')
+packet_definition = xtcedef.XtcePacketDefinition(xtce_document)
+
+# Initialize parser with custom XTCE definition
+my_parser = parser.PacketParser(packet_definition)
+
+# Generator object that spits out parsed Packet objects containing header and data attributes
+packet_generator = my_parser.generator(binary_data)
+
+for packet in packet_generator:
+    # Do something with the packet data
+    print(packet.header)
```

Comment:
Just for my understanding, would the next step here be to put each field into a CDF?

Reply:
Sorry @sdhoyt . This PR has moved to https://github.com/IMAP-Science-Operations-Center/imap_processing/pull/1. 

But to answer your questions. That's my understanding too. I envision we unpack data using packet parser. In that step we don't do any modification to data. We just unpack data.

Then we perform additional changes as needed for each instrument. Some like to leave data as it is, some like to decompress data and format it to their requirements, and some may just dump data based on flags.